# Ejercicio1 - Utilizar operadores aritméticos

# Operadores aritméticos en Python

Python proporciona operadores aritméticos comunes para que puedas realizar operaciones matemáticas en tu código. Estos incluyen las cuatro operaciones principales de suma, resta, multiplicación y división.

Exploremos cómo podemos crear un programa que pueda calcular la distancia entre dos planetas. Comenzaremos usando dos distancias de planetas: Tierra (149.597.870 km) y Júpiter (778.547.200 km).

TIP Dedica unos minutos a tratar de encontrar una solución. Luego desplázate hacia abajo hasta la parte inferior para ver si has logrado compilar el programa de acuerdo con las especificaciones.

Nota: Quita las comas cuando uses los valores.

In [ ]:
# Crear variables para almacenar las dos distancias
# ¡Asegúrate de quitar las comas!

In [1]:
earth = 149597870; #distance from the earth to the sun in km

jupiter = 778547200; #distance from jupiter to the sun in km

In [3]:
#Distance between earth and jupiter in km

dist= jupiter - earth;

#tranform to miles
distMiles = dist*0.621;


628949330
390577533.93


# Realizar la operación

Con los valores obtenidos, es el momento de añadir el código para realizar la operación. Restarás el primer planeta del segundo para determinar la distancia en kilómetros. A continuación, puedes convertir la distancia del kilómetro en millas multiplicándola por 0.621.

# Prueba tu proyecto

Con el código creado, ejecuta el notebook para obtener el resultado. Deberías recibir lo siguiente:

    628949330
    390577534
Recuerda: Lo importante es intentarlo, puedes auxiliarte de internet, libros, etc. El objetivo de una kata es la comprobación de la comprensión de la lección y su aplicación inmediata a la resolución de problemas. Go for it!

In [6]:
print(dist);
print(round(distMiles));

628949330
390577534


# Ejercicio 2: convierte cadenas en números y usa valores absolutos

Crear una aplicación para trabajar con números y entrada de usuario
Con frecuencia, deberás convertir los valores de cadena en números para realizar correctamente diferentes operaciones o determinar el valor absoluto de un número.

Para crear nuestra aplicación, queremos leer la distancia del sol para dos planetas, y luego mostrar la distancia entre los planetas. Haremos esto usando input para leer los valores, int para convertir a entero y luego abs para convertir el resultado en su valor absoluto.

# Lee los valores
Usando input, agrega el código para leer la distancia del sol para cada planeta, considerando 2 planetas.

In [ ]:
# Almacenar las entradas del usuario
#Pista: variable = input("¿Cuál es tu nombre?")

In [7]:
print("We will determine the distance between two planets");
input1 = input("Type th distance of a planet to the sun: ");
input2 = input("Type th distance of another planet to the sun: ");

We will determine the distance between two planets
Type th distance of a planet to the sun57900000
Type th distance of another planet to the sun	149600000


# Convertir a número

Debido a que input devuelve valores de cadena, necesitamos convertirlos en números. Para nuestro ejemplo, usaremos int

In [ ]:
# Convierte las cadenas de ambos planetas a números enteros

In [8]:
planet1 = int(input1);
planet2 = int(input2);

# Realizar el cálculo y convertir a valor absoluto

Con los valores almacenados como números, ahora puedes agregar el código para realizar el cálculo, restando el primer planeta del segundo. Debido a que el segundo planeta podría ser un número mayor, usarás abs para convertirlo a un valor absoluto. También agregarás el código para mostrar el resultado en millas multiplicando la distancia del kilómetro por 0.621

In [10]:
# Realizar el cálculo y determinar el valor absoluto

distKm = abs(planet2 - planet1);

# Convertir de KM a Millas

distMiles = distKm*0.621;

print(distKm);
print(distMiles);

91700000
56945700.0


Prueba tu aplicación
Para probar el proyecto, ejecuta tu notebook. En la parte superior de vscode surgirá un cuadro de diálogo para que proporciones las distancias. Puede utilizar los datos de la tabla siguiente:

    Planeta	Distancia al sol
    Mercurio	57900000
    Venus	108200000
    Tierra	149600000
    Marte	227900000
    Júpiter	778600000
    Saturno	1433500000
    Urano	2872500000
    Neptuno	4495100000